src/mgr/ActivePyModules.cc

```diff
@@ -718,7 +718,7 @@ PyObject* ActivePyModules::get_counter_python(
       const auto &avg_data = counter_instance.get_data_avg();
       for (const auto &datapoint : avg_data) {
         f.open_array_section("datapoint");
-        f.dump_unsigned("t", datapoint.t.to_nsec());
+        f.dump_float("t", datapoint.t);
```

Comment:
with this change do we keep the same precision? does the decimal part contains the nano seconds precision?

Reply:
Short answer: no.

Long answer: Float in Python is actually double. A double has [~15 decimal digit precision](https://stackoverflow.com/questions/9999221/double-precision-decimal-places), and current UNIX time spends 10 decimal digits, so right now we can only keep timestamps with 10 us precision, until November 2286, when Ceph Mgr would pass to have 100 us precision. By then, [temperatures will have increased +12ºC](http://tool.globalcalculator.org/globcalc.html?levers=2211111e13be1111c2c2c1n31hfjdcef222hp233f211111fn2211111111/dashboard/en), so by then we'll probably don't care too much about this.

Reply:
:sweat_smile:

Reply:
Hehe, I was not clear with my question. Let me try to better explain my doubt:

Currently the `f.dump_unsigned("t", datapoint.t.to_nsec());` will output a value like `1568979338062425709`.
After this change would the output be: `1568979338.062425709` or `1568979338.000000000` ?

Reply:
Well, not exactly like this (binary decimals are weirder than base 10 decimals). It apparently has more than 15-16 digit precision, but it's IEEE 754 fake precision:
```py
>>> 1568979338062425709/1e9
1568979338.0624256

# if we increase 100 ns, we get 238 nsec difference
>>> (1568979338062425709 + 100)/1e9
1568979338.0624259
```

Reply:
Nevertheless, metrics stores like [Prometheus support millisecond precision](https://prometheus.io/docs/concepts/data_model/#samples).